### DISCUSSION TITLE
Installing IDAKLU on Windows without using WSL

### DISCUSSION THREAD

**Discussion by abhishek-appana** [pybamm-team/PyBaMM#4373] [Q&A]
Hello @agriyakhetarpal ,

Thank you for your instructions for installing IDAKLU on Windows in this [discussion ](https://github.com/pybamm-team/PyBaMM/discussions/4296).

I followed them to install IDAKLU without using WSL. When I run `pip install -ve .` after completing all the steps, PyBaMM gets installed but the following error comes up in the log

```
Building wheels for collected packages: pybamm
  Building editable for pybamm (pyproject.toml): started
  Running command Building editable for pybamm (pyproject.toml)
  Traceback (most recent call last):
    File "<frozen runpy>", line 198, in _run_module_as_main
    File "<frozen runpy>", line 88, in _run_code
    File "C:\Users\rajaa\AppData\Local\Programs\Python\Python311\Scripts\cmake.exe\__main__.py", line 4, in <module>
  ModuleNotFoundError: No module named 'cmake'

``` 

I get the following when I try to use IDAKLU solver via VS code.
```
ImportError: KLU is not installed
``` 

Can someone let me know if I'm doing something wrong?

Thanks in advance!

**Comment by agriyakhetarpal** [comment]
Hi @abhishek-appana, you will need to install CMake on your machine to be able to compile the solver. You can get it with any Windows-based package manager of your choice, such as Chocolatey, WinGet, NuGet, Scoop, and nowadays using `pip`/`pipx`.

In a virtual environment,

```pwsh
pip install cmake
```

should work well

P.S. I wouldn't recommend installing the CMake you get from `pip` outside a virtual environment, though – it is designed to make working with Python extension modules and their compilation and isn't meant to replace a global installation of CMake (see the upstream repository at https://github.com/scikit-build/cmake-python-distributions).

**Reply by abhishek-appana** [reply]
Thank you for your response @agriyakhetarpal . I just did that. I am still getting the same error.

Here is the output from `pip list` inside the virtual environment terminal

<details><summary>Details</summary>
<p>

Package            Version     Editable project location
------------------ ----------- ----------------------------------------------------------------------------------------------------------------
anytree            2.12.1
casadi             3.6.6
certifi            2024.7.4
charset-normalizer 3.3.2
cmake              3.30.2
24.1 urllib3-2.2.2 xarray-2024.7.0
PS C:\Users\rajaa\Abhishek\PyBaMM> pip list
Package            Version     Editable project location
------------------ ----------- ----------------------------------------------------------------------------------------------------------------
anytree            2.12.1
casadi             3.6.6
certifi            2024.7.4
charset-normalizer 3.3.2
cmake              3.30.2
idna               3.8
PS C:\Users\rajaa\Abhishek\PyBaMM> pip list
Package            Version     Editable project location
------------------ ----------- ----------------------------------------------------------------------------------------------------------------
anytree            2.12.1
casadi             3.6.6
certifi            2024.7.4
charset-normalizer 3.3.2
cmake              3.30.2
idna               3.8
Package            Version     Editable project location
------------------ ----------- ----------------------------------------------------------------------------------------------------------------
anytree            2.12.1
casadi             3.6.6
certifi            2024.7.4
charset-normalizer 3.3.2
cmake              3.30.2
idna               3.8
anytree            2.12.1
casadi             3.6.6
certifi            2024.7.4
charset-normalizer 3.3.2
cmake              3.30.2
idna               3.8
cmake              3.30.2
idna               3.8
mpmath             1.3.0
numpy              1.26.4
idna               3.8
mpmath             1.3.0
numpy              1.26.4
packaging          24.1
mpmath             1.3.0
numpy              1.26.4
packaging          24.1
numpy              1.26.4
packaging          24.1
pandas             2.2.2
pip                24.2
packaging          24.1
pandas             2.2.2
pip                24.2
pandas             2.2.2
pip                24.2
platformdirs       4.2.2
pip                24.2
platformdirs       4.2.2
pooch              1.8.2
platformdirs       4.2.2
pooch              1.8.2
pybamm             24.5        C:\Users\rajaa\Abhishek\PyBaMM
pooch              1.8.2
pybamm             24.5        C:\Users\rajaa\PyBaMM
pybamm             24.5        C:\Users\rajaa\PyBaMM
python-dateutil    2.9.0.post0
pytz               2024.1
requests           2.32.3
scipy              1.14.1
setuptools         65.5.0
six                1.16.0
sympy              1.13.2
typing_extensions  4.12.2
tzdata             2024.1
urllib3            2.2.2
xarray             2024.7.0

</p>
</details> 

Here is the error I got when I ran `pip install -ve .`

```
Building wheels for collected packages: pybamm
  Running command Building editable for pybamm (pyproject.toml)
  Traceback (most recent call last):
    File "<frozen runpy>", line 198, in _run_module_as_main
    File "<frozen runpy>", line 88, in _run_code
    File "C:\Users\rajaa\Abhishek\PyBaMM\src\pybamm\.venv\Scripts\cmake.exe\__main__.py", line 4, in <module>
  ModuleNotFoundError: No module named 'cmake'
```
